public/app/plugins/panel/bargauge/BarGaugePanel.tsx

```diff
@@ -22,14 +24,19 @@ export class BarGaugePanel extends PureComponent<PanelProps<BarGaugeOptions>> {
     const { field, display, view, colIndex } = value;
     const { openMenu, targetClassName } = menuProps;
 
+    let processor: DisplayProcessor | undefined = undefined;
+    if (view && isNumber(colIndex)) {
+      processor = view!.getFieldDisplayProcessor(colIndex as number);
```

Comment:
nit: might be an idea to make the `getFieldDisplayProcessor` a little bit more generous in what it can accept as a param. Then we can move this check to that function instead of needing to do it on call site.